src/charm.py

```diff
@@ -318,7 +325,7 @@ def set_secret(self, scope: Scopes, key: str, value: str | None) -> str | None:
             return None
         secret_key = self._translate_field_to_secret_key(key)
         # Old translation in databag is to be deleted
-        self.peer_relation_data(scope).delete_relation_data(peers.id, [key])
+        self.scoped_peer_data(scope).pop(key, None)
         self.peer_relation_data(scope).set_secret(peers.id, secret_key, value)
```

Comment:
Secrets caching breaks when we try to reset the secret to the same value. `set_secret()` will not go through if it is the same value, but `delete_relation_data()` will go through and set the secret to None. Just clearing the peer data for potential legacy value should be sufficient here.